libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -109,10 +110,7 @@ bool ByteBuffer::advance(uint32_t n)
     return true;
 }
 
-/*
-  read len bytes without advancing the read pointer
- */
-uint32_t ByteBuffer::peekbytes(uint8_t *data, uint32_t len)
+int ByteBuffer::peekiovec(ByteBuffer::IoVec iovec[2], uint32_t len)
 {
     if (len > available()) {
         len = available();
```

Comment:
Since you already need 'n', I would start the code with:
```
uint32_t n = available();
if (len > n) len = n;
```
 and have available() only called once when first if is true (without the need to count on compiler optimizations)